rclpy/test/test_logging_rosout.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from rcl_interfaces.msg import Log
+import rclpy
+from rclpy.executors import SingleThreadedExecutor
+
+TEST_PARAMETERS = [
+    ('enable_global_rosout_enable_node_rosout', True, True, True),
+    ('enable_global_rosout_disable_node_rosout', True, False, False),
+    ('disable_global_rosout_enable_node_rosout', False, True, False),
+    ('disable_global_rosout_disable_node_rosout', False, False, False),
+]
+
+raw_subscription_msg = None  # None=No result yet
+
+
+def raw_subscription_callback(msg):
+    global raw_subscription_msg
+    raw_subscription_msg = msg
+
+
+@pytest.mark.parametrize(
+    'name,enable_global_rosout_logs,enable_node_rosout,expected_data',
+    TEST_PARAMETERS)
+def test_enable_rosout(
+    name,
+    enable_global_rosout_logs,
+    enable_node_rosout,
+    expected_data
+):
+    if enable_global_rosout_logs:
+        args = ['--ros-args', '--enable-rosout-logs']
+    else:
+        args = ['--ros-args', '--disable-rosout-logs']
+
+    context = rclpy.context.Context()
+    rclpy.init(context=context, args=args)
+    executor = SingleThreadedExecutor(context=context)
+
+    # create node
+    node = rclpy.create_node(
+        node_name='my_node_'+name,
+        namespace='/my_ns',
+        enable_rosout=enable_node_rosout,
+        context=context
+    )
+    executor.add_node(node)
+
+    global raw_subscription_msg
+    raw_subscription_msg = None
+    # create subscriber of 'rosout' topic
+    node.create_subscription(
+        Log,
+        'rosout',
+        raw_subscription_callback,
+        1,
+        raw=True
+    )
+
+    node.get_logger().info('SOMETHING')
+    executor.spin_once(timeout_sec=1)
```

Comment:
@ivanpauno (tagging you because you reviewed it), watch out for things like this. It's really fragile, we're seeing this test fail a lot either due to the timeout before the event or because spinning once does not guarantee that the event you wanted handled was the one that would be handled...

Reply:
For example: https://ci.ros2.org/job/ci_osx/8540/testReport/rclpy.src.ros2.rclpy.rclpy.test/test_logging_rosout/test_enable_rosout_enable_global_rosout_enable_node_rosout_True_True_True_/

Reply:
@wjwwood 

our bad, sorry about that.
we will submit issue against [problem](https://ci.ros2.org/job/ci_osx/8540/testReport/rclpy.src.ros2.rclpy.rclpy.test/test_logging_rosout/test_enable_rosout_enable_global_rosout_enable_node_rosout_True_True_True_/) and try to fix it.

Reply:
Thanks @wjwwood.
I overlooked that when reviewing, but the test is definitely flaky.

I've created an issue https://github.com/ros2/rclpy/issues/546.

Reply:
@fujitatomoya if you're not planning to work on a fix in a near future, let me know and I will work on it. Thanks!

Reply:
i think i can work on this early next week, but if you already have idea, you can go ahead.

Reply:
@wjwwood 

Thank you for pointing out my mistake. I will fix it.

@ivanpauno @fujitatomoya

It's my fault. I will fix this problem tomorrow in #546.

Reply:
No worries, I wasn't trying to call anyone out, just wanted to point it out before I forgot and explain why it is fragile.